hangman.py

```diff
@@ -113,19 +145,54 @@ def hangman(secret_word):
     
     Follows the other limitations detailed in the problem write-up.
     '''
-    # FILL IN YOUR CODE HERE AND DELETE "pass"
-    pass
-
-
-
-# When you've completed your hangman function, scroll down to the bottom
-# of the file and uncomment the first two lines to test
-#(hint: you might want to pick your own
-# secret_word while you're doing your own testing)
-
-
-# -----------------------------------
-
+    print('''
+    __  __                                           ___ ____ 
+   / / / /___ _____  ____ _____ ___  ____ _____     <  // __ \\
+  / /_/ / __ `/ __ \/ __ `/ __ `__ \/ __ `/ __ \    / // / / /
+ / __  / /_/ / / / / /_/ / / / / / / /_/ / / / /   / // /_/ / 
+/_/ /_/\__,_/_/ /_/\__, /_/ /_/ /_/\__,_/_/ /_/   /_(_)____/  
+                  /____/                                      
+''')
+    print(f'I am thinking of a word that is {len(secret_word)} letters long.')
+    guesses = 6
+    letters_guessed = []
+    vowels = ('a', 'e', 'i', 'o', 'u')
+    warnings = 3
+    while 1:
+        if is_word_guessed(secret_word, letters_guessed):
+            print(victory(guesses, secret_word))
+            break
+        elif guesses <= 0:
+            print(f'\033[7mYou lost. Guesses exceeded. The secret word was - {secret_word}\033[0m')
+            break
+        else:
+            print(get_guessed_word(secret_word, letters_guessed))
+            print(f'You have {guesses} guesses left.')
+            available = get_available_letters(letters_guessed)
+            print(f'Available letters: {available}')
+            while 1:
+                letter = input('Enter your guess letter:   ')
+                if letter.isalpha():
+                    break
+                else:
+                    if warnings <= 0:
+                        guesses -= 1
+                        if guesses == 0:
+                            break
+                    else:
+                        warnings -= 1
+                    print(
+                        f'\033[31mYou must enter a letter of the Latin alphabet. Not a number or symbol. {warnings} warnings left\033[0m')
+            letters_guessed.append(letter.lower())
```

Comment:
Can you remember that you need also check whether that letter was entered before? And print appropriate message